### PR TITLE
HSEARCH-3480 Rename fromExternalRepresentation() to of() in enums

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/HSEARCH-3480

Maybe don't merge it just yet but incorporate it in our private branch so that we can test that everything is working fine before merging.

Note that I also changed things to consistently normalize the input values and log the normalized values as it's the one compared with the accepted values.

### CODE REVIEW

**Comment by yrodiere** [comment]
Just pushed a commit to use the same implementation everywhere, as discussed over the chat.

File: backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/logging/impl/Log.java [hunk 1]
```diff
@@ -297,9 +299,9 @@ SearchException cannotMixElasticsearchSearchTargetWithOtherBackend(IndexSearchTa
 			value = "Index '%1$s' requires multi-tenancy but the backend does not support it in its current configuration.")
 	SearchException multiTenancyRequiredButNotSupportedByBackend(String indexName, @Param EventContext context);
 
-	@Message(id = ID_OFFSET_3 + 15,
-			value = "Unknown multi-tenancy strategy '%1$s'.")
-	SearchException unknownMultiTenancyStrategyConfiguration(String multiTenancyStrategy);
+	@Message(id = ID_OFFSET_3 + 15, value = "Invalid multi-tenancy strategy name: '%1$s'."
+			+ " Valid names are: %2$s.")
```
Comment (gsmet) [inline backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/logging/impl/Log.java]: maybe valid strategies?

File: backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/logging/impl/Log.java [hunk 2]
```diff
@@ -455,4 +457,8 @@ SearchException conflictingIdentifierTypesForPredicate(ToDocumentIdentifierValue
 
 	@Message(id = ID_OFFSET_3 + 51, value = "The operation was skipped due to the failure of a previous work in the same changeset.")
 	SearchException elasticsearchSkippedBecauseOfPreviousWork(@Cause Throwable skippingCause);
+
+	@Message(id = ID_OFFSET_3 + 52, value = "Invalid index lifecycle strategy name: '%1$s'."
+			+ " Valid names are: %2$s.")
```
Comment (gsmet) [inline backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/logging/impl/Log.java]: Same here.

File: backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/logging/impl/Log.java (hunk 1 above)
Comment (yrodiere) [reply backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/logging/impl/Log.java]: It looks better to me this way, but feel free to change it if you feel strongly against it.

**Comment by yrodiere** [comment]
I rebased and squashed this PR with changes you made in another branch.
The build passes, so I'll merge. Thanks!

**Pull request merged**